server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSource.java

```diff
@@ -179,6 +180,17 @@ public FieldData(IndexFieldData<?> indexFieldData) {
             public SortedBinaryDocValues bytesValues(LeafReaderContext context) {
                 return indexFieldData.load(context).getBytesValues();
             }
+
+            public static class RangeFieldData extends FieldData {
```

Comment:
The type hierarchy in ValuesSource is a little arcane, and I just picked something that looked reasonable.

Reply:
Having just done some more thinking on the `ValuesSource` refactor, I'm in agreement that Range should be a peer to `Numeric`, `GeoPoint`, etc.  I'll wire that up shortly.